### PR TITLE
feat(Richup.io): add presence

### PR DESCRIPTION
**Pull request by eggsy**
## Description 
Presence for [Richup.io](https://richup.io), a new online Monopoly-like fun game. 

**Note to verifiers:** The website is a compiled React app which (I believe) was styled using CSS modules, since this results in random class names and since there were no IDs (or basically any other unique identifiers) for elements, I had to get data using ugly query selectors. I hope that doesn't create problems, I tried to be as concise as possible and tested everything. 

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

SCR-20230126-2wu
SCR-20230126-2x0

</details>

### CODE REVIEW

File: websites/R/Richup.io/presence.ts [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+let startTimestamp: number;
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "rio-logo",
+		},
+		{ pathname, href } = location,
+		settings = {
+			showJoinRoomButton: await presence.getSetting("join-room-button"),
+			showProfileButton: await presence.getSetting("profile-button"),
```
Comment (DarkVillager) [inline websites/R/Richup.io/presence.ts:22]: no, do it like this.
https://github.com/PreMiD/Presences/blob/ce2965e9aea36106f4fe16398ecad2c3dcea9aa4/websites/T/Tesla/presence.ts#L7-L12

**Review by DarkVillager — CHANGES_REQUESTED** [review]
?

File: websites/R/Richup.io/presence.ts (hunk 1 above)
Comment (eggsy) [reply websites/R/Richup.io/presence.ts:22]: alright

File: websites/R/Richup.io/presence.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+		if (showProfileButton === true) {
+			presenceData.buttons = [
+				{
+					label: "View Profile",
+					url: href,
+				},
+			];
+		}
+	} else if (pages[pathname]) {
+		presenceData.details = "Viewing a page";
```
Comment (DarkVillager) [inline websites/R/Richup.io/presence.ts:87]: https://github.com/PreMiD/Localization/blob/079bb7fb96b27c76e5eb6d1b5777e78433a34f82/src/Presence/general.json#L294-L297

**Comment by Antwain-Jordan** [comment]
Are there any updates for this rich presence?

**Comment by DarkVillager** [comment]
> Are there any updates for this rich presence?

Patience.

**Review by EncryptedDev — DISMISSED** [review]

File: websites/R/Richup.io/metadata.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.7",
+  "service": "Richup.io",
+  "author": {
+    "id": "162969778699501569",
+    "name": "EGGSY#3388"
```
Comment (theusaf) [inline websites/R/Richup.io/metadata.json:6]: Discriminator added by old version of pmd.
```suggestion
    "name": "EGGSY"
```
Also, this file is not formatted properly. Use `yarn format` to reformat it.

File: websites/R/Richup.io/presence.ts [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+		"/store/upgrades": "Upgrades",
+		"/store/avatars": "Profile Pictures",
+		"/store/coins": "Coins",
+	};
+
+let startTimestamp: number;
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "rio-logo",
```
Comment (theusaf) [inline websites/R/Richup.io/presence.ts:17]: Imgur links are preferred over Discord asset keys now.
```suggestion
			largeImageKey: "https://i.imgur.com/4Nw3mle.png",
```

File: package-lock.json [hunk 5]
```diff
@@ -11244,7 +11244,7 @@
 		"pmd": {
 			"version": "git+ssh://git@github.com/PreMiD/pmd.git#a9f4ecf8c675a45ceab9936ca1cd9aae49975339",
 			"dev": true,
-			"from": "pmd@github:PreMiD/pmd#1.1.1",
+			"from": "pmd@github:PreMiD/pmd#1.1.2",
```
Comment (theusaf) [inline package-lock.json:11247]: Undo changes to lockfile

File: websites/R/Richup.io/presence.ts [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+		}
+
+		if (!startTimestamp) startTimestamp = Date.now();
+		presenceData.startTimestamp = startTimestamp;
+
+		const gameSettings = document
+			.querySelector("#app > div > div > div:nth-child(3)")
+			?.textContent?.includes("Game settings");
+
+		if (showJoinRoomButton === true && gameSettings === true) {
```
Comment (theusaf) [inline websites/R/Richup.io/presence.ts:64]: Unnecessary comparison of boolean variables.
```suggestion
		if (showJoinRoomButton && gameSettings) {
```

File: websites/R/Richup.io/presence.ts [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+			latestInfo = latestInfo.replace(pathname.split("/")[2], "");
+
+			/* Replace player names */
+			for (const player of players)
+				latestInfo = latestInfo.replace(new RegExp(player, "g"), "A player");
+
+			presenceData.state = latestInfo ? `🎲 ${latestInfo}` : "";
+		}
+
+		if (!startTimestamp) startTimestamp = Date.now();
```
Comment (theusaf) [inline websites/R/Richup.io/presence.ts:57]: Discord timestamps are in seconds.
```suggestion
		if (!startTimestamp) startTimestamp = Date.now() / 1000;
```

File: websites/R/Richup.io/presence.ts [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+				},
+			];
+		}
+	} else if (pathname.includes("/profile/")) {
+		presenceData.details = "Viewing a profile";
+		presenceData.state = document.querySelector(
+			"header > div > :last-child > span"
+		).textContent;
+
+		if (showProfileButton === true) {
```
Comment (theusaf) [inline websites/R/Richup.io/presence.ts:78]:
```suggestion
		if (showProfileButton) {
```

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/R/Richup.io/presence.ts (hunk 4 above)
Comment (eggsy) [reply websites/R/Richup.io/presence.ts:17]: weird but ok

File: websites/R/Richup.io/metadata.json (hunk 3 above)
Comment (eggsy) [reply websites/R/Richup.io/metadata.json]: now you got 4 more files to review :>

**Review by theusaf — APPROVED** [review]

**Comment by eggsy** [comment]
i thought you had write access bro 💀

**Review by Bas950 — APPROVED** [review]

**Comment by eggsy** [comment]
Come on guys it's been 2 weeks @EncryptedDev 😩

**Comment by Antwain-Jordan** [comment]
> Come on guys it's been 2 weeks @EncryptedDev 😩

Updates?

**Comment by RisingSunLight42** [comment]
Be PATIENT please, if there's an update, you'll be notified.

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**